### PR TITLE
ci: added conventional commits check 

### LINKED ISSUE #1366: Convention for commit messages
## Summary
The changelog for releases of Clio could benefit from better visibility of bugs vs. features, api breakage, etc. 
To allow that we could adopt some convention for our commit messages (squashed). 

[Here](https://gist.github.com/qoomon/5dfcdf8eec66a051ecd85625518cfd13) is an example to get inspiration from.

## Motivation
It would help us generate better changelogs.

## Solution
Create a convention document and follow it when squashing commits.

### PR DESCRIPTION
**Pull request by kuznetsss**
Fixes #1366

### CODE REVIEW

File: .github/workflows/clang-tidy.yml
Comment (godexsoft) [inline .github/workflows/clang-tidy.yml:112]: Need to adjust `clang-tidy_on_fix_merged.yml` accordingly if changing this title 👍

**Review by godexsoft — CHANGES_REQUESTED** [review]

**Review by godexsoft — DISMISSED** [review]

**Comment by kuznetsss** [comment]
`Check PR title` works fine, but it is failing because it doesn't have write access to the repo. It will be fixed once the PR is merged.

**Review by godexsoft — APPROVED** [review]

**Pull request merged**
